offchainservice/src/main/java/pl/edu/agh/blockchain/offchainservice/service/RegistrationService.java

```diff
@@ -0,0 +1,113 @@
+package pl.edu.agh.blockchain.offchainservice.service;
+
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import pl.edu.agh.blockchain.offchainservice.dto.PendingVerificationDTO;
+import pl.edu.agh.blockchain.offchainservice.dto.RegisterInformationDTO;
+import pl.edu.agh.blockchain.offchainservice.exceptions.*;
+import pl.edu.agh.blockchain.offchainservice.model.PendingVerification;
+import pl.edu.agh.blockchain.offchainservice.model.User;
+import pl.edu.agh.blockchain.offchainservice.repository.EmailRepository;
+import pl.edu.agh.blockchain.offchainservice.repository.PendingRepository;
+import pl.edu.agh.blockchain.offchainservice.utils.CodeGenerator;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class RegistrationService {
+
+    private final EmailRepository emailRepository;
+
+    private final PendingRepository pendingRepository;
+
+    private final EMailService eMailService;
+
+    private final DeploymentService deploymentService;
+
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    public RegistrationService(EmailRepository emailRepository, EMailService eMailService,
+                               PendingRepository pendingRepository, DeploymentService deploymentService) {
+        this.emailRepository = emailRepository;
+        this.eMailService = eMailService;
+        this.pendingRepository = pendingRepository;
+        this.deploymentService = deploymentService;
+    }
+
+    public void verifyUser(String email) {
+        if (!email.endsWith("@agh.edu.pl") && !email.endsWith("@student.agh.edu.pl")) {
+            throw new InvalidEmailDomainException(email + " is not AGH email");
+        }
+        if (!emailRepository.existsByEmail(email)) {
```

Comment:
I think we also need to check if an entry with such email exists in the `pending` table (user with this email received a verification token inside last 15 mins). If yes, then return some message like 'You already got verification token. Check your email'. Otherwise, continue sending email as now